src/util/kaldi-table-inl.h

```diff
@@ -340,6 +441,13 @@ template<class Holder>  class SequentialTableReaderScriptImpl:
     // or the user called FreeCurrent().. note,
     // if when called by user we are in this state,
     // it means the user called FreeCurrent().
+    kObjectLoaded, // Called NextScpLine() and found we already  yes         yes
```

Comment:
I don't think there is a need for a separate state kObjectLoaded because it behaves the same as kLoadSucceeded.  I.e. it means we have the object.  It complicates the analysis, IMO.